src/relayserver/RelayServer.ts

```diff
@@ -40,8 +40,16 @@ const GAS_RESERVE = 100000
 export class RelayServer extends EventEmitter {
   lastScannedBlock = 0
   lastRefreshBlock = 0
-  ready = false
-  lastWorkerFinished = Date.now()
+
+  // setReadyState set the time when the relayer becomes ready. negative= not ready
+  // Nan = not ready.
+  // Zero is already ready Since date.now is bigger)
+  // now+timeout - will be ready after timeout.
+  futureReadyTime = NaN
+
+  // if deferReadiness is true, then setReadyState(true) takes effect only after relayTimeout.
+  // it is set only after worker exception (=probably some provider error)
+  deferReadiness = false
```

Comment:
Ok, I am not approving this. This is a patchwork and it does not solve the issue.

Ready flag should stay there - it indicates balances and registration is going well. It should not be abused to also mean 'Infura connection is good' - lets create a separate mechanism for that.

Also, I don't see a need for 'deferReadiness' flag.  It never becomes 'false' again, so it is basically one-time thing.

The 'futureReadyTime' is also problematic IMO.

How about that - we have a 'successful handler runs' variable, init it with, IDK, `requiredSuccessfulHandlerRuns == 10`.

On a single thrown exception it is reset to 0, on a successful run it is incremented. Also if the worker does not complete before `workerTimeout` is executed.

Only consider relay to be 'ready' if it is 
a) old code `ready === true`
b) `this.successfulHandlerRuns > this.config.requiredSuccessfulHandlerRuns`